daemon/api_console_conf.go

```diff
@@ -0,0 +1,88 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/snapcore/snapd/overlord/auth"
+)
+
+var (
+	routineConsoleConfStartCmd = &Command{
+		Path: "/v2/internal/console-conf-start",
+		POST: consoleConfStartRoutine,
+	}
+)
+
+type consoleConfRoutine struct{}
+
+// ConsoleConfStartRoutineResult is the result of running the console-conf start
+// routine..
+type ConsoleConfStartRoutineResult struct {
+	ActiveSnapAutoRefreshChanges []string `json:"active-snap-refreshes,omitempty"`
+}
+
+func consoleConfStartRoutine(c *Command, r *http.Request, _ *auth.UserState) Response {
+	// no body expected, error if we were provided anything
+	defer r.Body.Close()
+	var routineBody interface{}
```

Comment:
Hmm, what's the expected decoding? I was expecting an empty struct, not an empty interface.

Reply:
ah yes an empty struct makes more sense, I just happened to do an empty interface here since there is expected to not be any body included in the post